### PR TITLE
Selenium 4 changes

### PR DESCRIPTION
**Pull request by RutvikChandla**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.browserstack.app</groupId>
+  <artifactId>java-browserstack</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <name>java-browserstack</name>
+  <!-- FIXME change it to the project's website -->
+  <url>http://www.example.com</url>
```
Comment (karanshah-browserstack) [inline pom.xml]: browserstack.com instead of example.com

File: src/test/java/com/browserstack/app/JavaLocalSample.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import java.time.Duration;
+import java.util.HashMap;
+
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import com.browserstack.local.Local;
+
+public class JavaLocalSample {
+  public static final String AUTOMATE_USERNAME = "rutvikchandla_2MEern";
+  public static final String AUTOMATE_ACCESS_KEY = "AXHzyg34Qr81Nep231pu";
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/app/JavaLocalSample.java]: Remove username and accesskey

File: src/test/java/com/browserstack/app/JavaLocalSample.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	//Stop the Local instance
+	bsLocal.stop();
+   driver.quit();
+ }
+// This method accepts the status, reason and WebDriver instance and marks the test on BrowserStack
+public static void markTestStatus(String status, String reason, WebDriver driver) {
+ final JavascriptExecutor jse = (JavascriptExecutor) driver;
+ jse.executeScript("browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\": \""+ status + "\", \"reason\": \"" + reason + "\"}}");
+}
+}
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/app/JavaLocalSample.java]: EOL

File: src/test/java/com/browserstack/app/JavaParallelSample.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+class DeviceTwo implements Runnable {
+	public void run() {
+		Hashtable<String, String> capsHashtable = new Hashtable<String, String>();
+		capsHashtable.put("deviceName", "Samsung Galaxy S20");
+		capsHashtable.put("realMobile", "true");
+		JavaParallelSample deviceTwo = new JavaParallelSample();
+		deviceTwo.executeTest(capsHashtable, "Thread 2");
+	}
+}
+class DeviceThree implements Runnable {
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/app/JavaParallelSample.java]: Can we create this dynamically? This looks very odd and repeated code.

File: src/test/java/com/browserstack/app/JavaSample.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.JavascriptExecutor;
+import java.net.URL;
+import java.time.Duration;
+import java.util.HashMap;
+
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+public class JavaSample {
+  public static final String AUTOMATE_USERNAME = "rutvikchandla_2MEern";
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/app/JavaSample.java]: Remove username and accesskey

File: src/test/java/com/browserstack/app/JavaSample.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+       markTestStatus("failed", "Some elements failed to load", driver);
+     }
+   driver.quit();
+ }
+// This method accepts the status, reason and WebDriver instance and marks the test on BrowserStack
+public static void markTestStatus(String status, String reason, WebDriver driver) {
+ final JavascriptExecutor jse = (JavascriptExecutor) driver;
+ jse.executeScript("browserstack_executor: {\"action\": \"setSessionStatus\", \"arguments\": {\"status\": \""+ status + "\", \"reason\": \"" + reason + "\"}}");
+}
+}
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/app/JavaSample.java]: EOL

**Review by karanshah-browserstack — CHANGES_REQUESTED** [review]

File: README copy.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+## <b> java-browserstack </b>
+
+A base language repo for working with selenium and browserstack.
+
+---
```
Comment (karanshah-browserstack) [inline README copy.md]: Rename file to README.md

**Review by karanshah-browserstack — CHANGES_REQUESTED** [review]

**Comment by RutvikChandla** [comment]
@karanshah-browserstack have done the changes please review.

File: src/test/java/com/browserstack/app/JavaLocalSample.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+	capabilities.setCapability("bstack:options", browserstackOptions);
+	capabilities.setCapability("sessionName", "BStack-[Java] Sample Test"); // test name
+	capabilities.setCapability("buildName", "BStack Local Build Number 1"); // CI/CD job or build name
+ 
+//Creates an instance of Local
+Local bsLocal = new Local();
+
+// You can also set an environment variable - "BROWSERSTACK_ACCESS_KEY".
+HashMap<String, String> bsLocalArgs = new HashMap<String, String>();
+bsLocalArgs.put("key", "AXHzyg34Qr81Nep231pu");
```
Comment (karanshah-browserstack) [inline src/test/java/com/browserstack/app/JavaLocalSample.java]: Remove this

**Review by karanshah-browserstack — CHANGES_REQUESTED** [review]

File: src/test/java/com/browserstack/app/JavaLocalSample.java (hunk 8 above)
Comment (RutvikChandla) [reply src/test/java/com/browserstack/app/JavaLocalSample.java]: Resolved

**Review by karanshah-browserstack — APPROVED** [review]

**Pull request merged**